src/components/Map/Map.js

```diff
@@ -24,6 +24,7 @@ export default function Map() {
   const [anchorEl, setAnchorEl] = useState(null);
   const [isMapFetching, setIsMapFetching] = useState(false);
   const [vpnAlert, setVpnAlert] = useState(true);
+  const [serverError, setServerError] = useState(false);
```

Comment:
The idea of using 2 different states for handling this situation is nice. But the implementation is basically a hack. I think `vpnAlert` default value should be `false` and at the same time that you set `serverError`, set the `vpnAlert` as well. 

The other solution can be using `useEffect` which depends on `serverError` and sets `vpnAlert` accordingly. But this one has performance downsides.

Although the solution seems to work, at least for now.

Reply:
in fact first I've thought about the util to check the Iran ip and accordingly show proper message to user but on that time of coding the priority was on speed of work.